bot/exts/evergreen/wikipedia.py

```diff
@@ -39,31 +39,42 @@ async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List
         async with self.bot.http_session.get(url=url) as resp:
             if resp.status == 200:
                 raw_data = await resp.json()
-                number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
+                if raw_data.get("query", None) is None:
+                    if raw_data.get("errors", None) is not None:
+                        log.info("There was an error regarding the Wikipedia API query.")
+                    else:
+                        log.info("There was an issue when trying to communicate with the Wikipedia API")
 
-                else:
                     await channel.send(
-                        "Sorry, we could not find a wikipedia article using that search term."
-                    )
+                        "There was an issue processing your Wikipedia request, please try again later.")
                     return
+                else:
```

Comment:
As for the first part, I think that wouldn't work since as the docs says the Wikipedia API "usually" returns a 200 status code even for errors and warnings, [(Docs)](https://www.mediawiki.org/wiki/API:Errors_and_warnings) so checking the status is actually not useful. I just left it there in case the API goes down.
As for the second part, yes, I'm not sure why I left it there! 😄

Reply:
Also, the API should never even send back an error, just in case of malformed query URLs, and I'm not sure why this error was raised in the first place, I tested out the payload that caused the error, but I could only reproduce it once.